src/libraries/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -443,325 +443,12 @@ private static void _BulkMoveWithWriteBarrier(ref byte destination, ref byte sou
     internal static partial class SpanHelpers
     {
         [Intrinsic] // Unrolled for small sizes
-        public static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
+        public static void ClearWithoutReferences(ref byte b, nuint byteLength)
         {
             if (byteLength == 0)
                 return;
 
-#if TARGET_AMD64 || TARGET_ARM64 || TARGET_LOONGARCH64
-            // The exact matrix on when ZeroMemory is faster than InitBlockUnaligned is very complex. The factors to consider include
-            // type of hardware and memory alignment. This threshold was chosen as a good balance across different configurations.
-            if (byteLength > 768)
-                goto PInvoke;
             Unsafe.InitBlockUnaligned(ref b, 0, (uint)byteLength);
```

Comment:
Without the prior conditional, we cannot truncate `byteLength` from `nuint` to `uint`.

Reply:
Correct, I reverted this change, but forgot to push 🙂